src/compute/arity.rs

```diff
@@ -88,7 +92,7 @@ where
 {
     if lhs.data_type() != rhs.data_type() {
```

Comment:
should we be consistent here and also not check this one?

Reply:
I was going to change that one with the Decimals PR, but now it can be done before merging this branch to Decimals